jax/_src/pallas/primitives.py

```diff
@@ -303,19 +303,25 @@ def _load_discharge_rule(in_avals, out_avals, *args_flat, args_tree, **_):
     raise NotImplementedError("Only one indexer supported in discharge rule.")
   idx = indexers[0]
   if all((isinstance(s, Slice) or not s.shape) for s in idx.indices):
-    # TODO(b/329733289): support strided load/store in interpret mode.
-    for s in idx.indices:
-      if isinstance(s, Slice) and s.stride > 1:
-        raise NotImplementedError("Unimplemented stride support.")
     indices = idx.indices
     scalar_dims = [not isinstance(s, Slice) and not s.shape for s in indices]
     slice_starts = [s.start if isinstance(s, Slice) else s for s in indices]
     slice_sizes = tuple(s.size if isinstance(s, Slice) else 1 for s in indices)
-    out_ones = lax.dynamic_slice(ref, slice_starts, slice_sizes=slice_sizes)
+    slice_strides = tuple(s.stride if isinstance(s, Slice) else 1 for s in indices)
+    indices = [start + lax.iota(jnp.int32, size) * stride for (start, size, stride)
```

Comment:
Agreed, it could be slow. The key advantage is that interpreter code (here) on host will produce the same behaviour as the compiled device code.

**Dynamic slice (or dynamic_update_slice) leads to incorrect behaviour** due to XLAs shifting rules. Noted as:
> the potentially surprising behavior for the case where the requested slice overruns the bounds of the array; in this case the start index is adjusted to return a slice of the requested size:
(https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.dynamic_slice.html)

an example is given in https://github.com/google/jax/issues/21143

The use of a gather instead of the dynamic slice fixes that incorrect behaviour